frontend/index.js

```diff
@@ -387,7 +390,9 @@ async function handleHashChange() {
   crumbs && updateBreadcrumbs(crumbs)
   clearOutput()
   await format()
-  editor.onUpdate(clearHash)
+  // Clear hash right away for the long ugly gzip/b64 content URL fragments,
+  // keep the fragment until the first edit for the rest.
+  opts.content ? clearHash : editor.onUpdate(clearHash)
```

Comment:
This didn't clear the hash. I think you needed `clearHash()` (with parens).

Reply:
argh :( thank you!!

Reply:
I'm sorry I didn't test that.

Reply:
The one down-side I see with this is that it the history.pushState mucks up the back-forth navigation with browser button, but maybe that's ok.

The other shitty thing I only notice now is that the menu bar doesn't sync with contents anymore. this needs more thinking / UX advice.

Reply:
re: history - I'm not sure what was intended exactly, but I don't think I would add to the history when clearing the hash. I think history should contain your navigation path generally and no navigation was done.

re: menu bar - sorry I don't understand this. What is the "menu bar" (hamburger?) and what is out of sync?

Reply:
re history: there is no option not to add a URL change to the history iiuc.
with `history.replaceState` the _last_ state gets replaced with the current one (e.g. no hash) and the previous entry, e.g. `#welcome` is "removed" from the history (not what we want) and with `history.pushState` you simple add a new entry. I think empty hash should be added to history and I don't actually understand how it couldn't be.

re menu bar: the bread crumbs, the thing on the top, like `Tour > Welcome`.

Reply:
HOK.
Long story short, I might have found a workaround. I'm not sure if this makes things better... it's all added to yet another last commit. please let me know what you think.

Reply:
maybe we need to discuss this in person. I understand your comment on the navigation path but I don't think this is an option unfortunately....

Reply:
I think it's all fine. I'm no longer sure what I was talking about. But having a play and the navigation seems fine.